src/ws/session.c

```diff
@@ -556,6 +609,10 @@ perform_tlscert (const char *rhost)
 
   res = open_session (pamh);
 
+  create_s4u_ticket (username);
```

Comment:
I think we need to talk about this a bit more explicitly: why not enable this also for password authentication or even Kerberos authentication?

Reply:
For Kerberos, no -- we already create a TGT there, which is much more useful (and *much* easier to use/configure) than the S4U one. For password auth, we could do this (at least if the machine is in a FreeIPA domain). I didn't do it here for the principle of least surprise, as we already use passwords to auth to sudo (and promote SSH keys), but it wouldn't hurt I guess.

Reply:
Sorry, I remember now: For password auth we already (implicitly) create a TGT, that's just how domain-managed users/PAM work:

![image](https://user-images.githubusercontent.com/200109/127658159-8c4e69cb-e4f6-418b-981e-c56cc797f4c3.png)

So an S4U one would again be a downgrade, and thus pointless IMHO.

Reply:
I guess we aren't super interested in the scenario where the user has a different password for krb than they do on the local machine.  That's kinda missing the point of what this is all supposed to be about...

This is cool: `KCM:256000500:68970`.  Any way we can steal that for our own cache instead of the tmpfile business?